lib/spack/docs/package_review_guide.rst

```diff
@@ -0,0 +1,388 @@
+All packages have a URL, though for some :ref:`build-systems` it is derived automatically and not visible in the package.
+
+Reasons :ref:`versions <versions-and-fetching>` may become invalid include:
+
+* the new URL does not support Spack version extrapolation;
+* the addition of or changes to ``url_for_version`` involve checks of the ``spec``'s version instead of the ``version`` argument or the (usually older) versions are not covered;
+* extrapolation of the derived URL no longer matches that of older versions; and
+* the older versions are no longer available.
+
+**Action.**
+Checking existing version directives with checksums can usually be done manually with the modified package using `spack checksum <https://spack.readthedocs.io/en/latest/command_index.html#spack-checksum>`_.
+
+**Solutions.**
+Options for resolving the problem that can be suggested for investigation depend on the source.
+
+In simpler cases involving ``url`` or ``url_for_version``, invalid versions can sometimes be corrected by ensuring all versions are covered by ``url_for_version``.
+Alternatively, especially for older versions, the version-specific URL can be added as an argument to the ``version`` directive.
+
+Sometimes the derived URLs of versions on the hosting system can vary.
+This commonly happens with Python packages.
+For example, the case of one or more letters in the package name may change at some point (e.g., `py-sphinx <https://github.com/spack/spack-packages/blob/develop/repos/spack_repo/builtin/packages/py_sphinx/package.py>`_).
+Also, dashes may be replaced with underscores (e.g., `py-scikit-build <https://github.com/spack/spack-packages/blob/develop/repos/spack_repo/builtin/packages/py_scikit_build/package.py>`_).
+In some cases, both changes can occur for the same package.
+As these examples illlustrate, it is sometimes possible to add a ``url_for_version`` method to override the default derived URL to ensure the correct one is returned.
+
+If older versions are no longer available and there is a chance someone has the package in a build cache, the usual approach is to first suggest :ref:`deprecating <deprecate>` them in the package.
+
+.. _vcs_url_reviews:
+
+``git``, ``hg``, ``svn``, or ``cvs``
+------------------------------------
+
+If the :ref:`repository-specific URL <vcs-fetch>` for fetching branches or the version control system (VCS) equivalent changes, there is a risk that the listed versions are no longer accessible.
+
+**Action.**
+You may need to check the new source repository to confirm the presence of all of the listed versions.
+
+.. _maintainers_reviews:
+
+``maintainers`` directive
+-------------------------
+
+**Action.**
+If the new package does not have a :ref:`maintainers <maintainers>` directive, ask the Contributor to add one.
+
+.. note::
+
+   This request is optional for existing packages.
+
+   Be prepared for them to refuse.
```

Comment:
I think this should be restructured to suggest putting less pressure on people to be maintainers.

```suggestion
.. _maintainers_reviews:

``maintainers`` directive
-------------------------

**Action.**
If the new package does not have a :ref:`maintainers <maintainers>` directive, suggest that the Contributor could list themself.

.. note::

   This request is optional. Be prepared for them to turn it down.
```